google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineStandardLibraryPanel.java

```diff
@@ -65,67 +60,47 @@ public AppEngineStandardLibraryPanel(boolean enabled) {
     // Objectify and Endpoints are dependencies of the App Engine API.
     objectifyCheckBox.setSelected(false);
     endpointsCheckBox.setSelected(false);
-    appEngineApiCheckBox.addItemListener(new ItemListener() {
-      @Override
-      public void itemStateChanged(ItemEvent event) {
-        JCheckBox checkbox = (JCheckBox) event.getItem();
-
-        if (!checkbox.isSelected()) {
-          objectifyCheckBox.setSelected(false);
-          endpointsCheckBox.setSelected(false);
-        }
+    appEngineApiCheckBox.addItemListener(event -> {
+      JCheckBox checkbox = (JCheckBox) event.getItem();
+
+      if (!checkbox.isSelected()) {
+        objectifyCheckBox.setSelected(false);
+        endpointsCheckBox.setSelected(false);
       }
     });
-    objectifyCheckBox.addItemListener(new AppEngineApiDependencyCheckboxListener());
-    endpointsCheckBox.addItemListener(new AppEngineApiDependencyCheckboxListener());
+    objectifyCheckBox.addItemListener(this::listenAppEngineApiDependencyCheckbox);
+    endpointsCheckBox.addItemListener(this::listenAppEngineApiDependencyCheckbox);
   }
 
   public Set<AppEngineStandardMavenLibrary> getSelectedLibraries() {
-    return Sets.newHashSet(Collections2.filter(
-        Collections2.transform(libraries,
-            new Function<JCheckBox, AppEngineStandardMavenLibrary>() {
-              @Nullable
-              @Override
-              public AppEngineStandardMavenLibrary apply(JCheckBox libraryCheckbox) {
-                return libraryCheckbox.isSelected()
-                    ? AppEngineStandardMavenLibrary
-                    .getLibraryByDisplayName(libraryCheckbox.getText())
-                    : null;
-              }
-            }),
-        Predicates.notNull()
-    ));
+    return libraries.stream()
+        .filter(JCheckBox::isSelected)
+        .map(
+            library ->
+                AppEngineStandardMavenLibrary.getLibraryByDisplayName(library.getText()).get())
+        .collect(toSet());
   }
 
   public void setSelectedLibraries(Set<AppEngineStandardMavenLibrary> mavenLibraries) {
-    Collection<String> availableLibraryNames = Collections2.transform(mavenLibraries,
-        new Function<AppEngineStandardMavenLibrary, String>() {
-          @Nullable
-          @Override
-          public String apply(AppEngineStandardMavenLibrary mavenLibrary) {
-            return mavenLibrary.getDisplayName();
-          }
-        });
-
-    for (JCheckBox libraryCheckbox : libraries) {
-      libraryCheckbox.setSelected(availableLibraryNames.contains(libraryCheckbox.getText()));
-    }
+    Set<String> availableLibraryNames = mavenLibraries.stream()
+        .map(AppEngineStandardMavenLibrary::getDisplayName)
+        .collect(toSet());
+
+    libraries.stream()
+        .filter(library -> availableLibraryNames.contains(library.getText()))
+        .forEach(checkbox -> checkbox.setSelected(true));
```

Comment:
I don't think this is the same behavior. Previously the logic was also unselecting libraries. Now, it looks like you are only ticking them.